### PR TITLE
Fix some IntPtr == null vs IntPtr.Zero checks

### PR DESCRIPTION
**Pull request by bartonjs**
I happened to spot one of these while working on one of my outstanding issues; then while trying to find it for a clean PR (remembering it involved Marshal.PtrToString) I saw a second.

cc: @stephentoub @eerhardt

### CODE REVIEW

**Comment by bartonjs** [comment]
```
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Net.Sockets.SafeCloseSocket.get_AsyncContext()
   at System.Net.Sockets.SocketPal.AcceptAsync(Socket socket, SafeCloseSocket handle, SafeCloseSocket acceptHandle, Int32 receiveSize, Int32 socketAddressSize, AcceptOverlappedAsyncResult asyncResult)
   at System.Net.Sockets.Socket.DoBeginAccept(Socket acceptSocket, Int32 receiveSize, AcceptOverlappedAsyncResult asyncResult)
   at System.Net.Sockets.Socket.BeginAccept(Socket acceptSocket, Int32 receiveSize, AsyncCallback callback, Object state)
   at System.Net.Sockets.Socket.BeginAccept(Int32 receiveSize, AsyncCallback callback, Object state)
   at System.Net.Sockets.Socket.BeginAccept(AsyncCallback callback, Object state)
   at System.Net.Sockets.Tests.SocketTestServerAPM.OnAccept(IAsyncResult result)
   at System.Net.LazyAsyncResult.Complete(IntPtr userToken)
   at System.Net.ContextAwareResult.CompleteCallback(Object state)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Net.ContextAwareResult.Complete(IntPtr userToken)
   at System.Net.LazyAsyncResult.ProtectedInvokeCallback(Object result, IntPtr userToken)
   at System.Net.LazyAsyncResult.InvokeCallback(Object result)
   at System.Net.Sockets.BaseOverlappedAsyncResult.CompletionCallback(Int32 numBytes, SocketError errorCode)
   at System.Net.Sockets.AcceptOverlappedAsyncResult.CompletionCallback(Int32 acceptedFileDescriptor, Byte[] socketAddress, Int32 socketAddressLen, SocketError errorCode)
   at System.Net.Sockets.SocketAsyncContext.AcceptOperation.InvokeCallback()
   at System.Net.Sockets.SocketAsyncContext.AsyncOperation.<QueueCompletionCallback>b__26(Object o)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
One or more tests failed while running tests from 'System.Net.Sockets.APMServer.Tests'.  Exit code 6.
```

While that sounds vaguely related, this code wouldn't be in it's codepath.

**Comment by bartonjs** [comment]
@dotnet-bot test this please

**Comment by bartonjs** [comment]
Opened #3948 for the test failure.

File: src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Crypto.cs [hunk 1]
```diff
@@ -98,7 +98,7 @@ internal static partial class Crypto
         internal static string GetX509RootStorePath()
         {
             IntPtr ptr = GetX509RootStorePath_private();
-            return ptr != null ?
+            return ptr != IntPtr.Zero ?
```
Comment (stephentoub) [inline src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Crypto.cs]: We shouldn't need these checks at all: PtrToStringAnsi is both implemented and documented to return null when passed IntPtr.Zero.

Comment (bartonjs) [inline src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Crypto.cs]: > We shouldn't need these checks at all: PtrToStringAnsi is both implemented and documented to return null when passed IntPtr.Zero.

Valid point, pushed out a replacement commit that removes the conditional expression altogether.

**Comment by stephentoub** [comment]
LGTM

**Pull request merged**